client/js/components/app.jsx

```diff
@@ -1,13 +1,50 @@
 import React from 'react';
 import Navbar from './navbar';
-
+import UserStore from '../stores/user-store';
 
 export default class App extends React.Component {
 
+  constructor() {
+    super();
+    this.state = { currentUser: UserStore.state.currentUser };
+
+    this._onChange = this._onChange.bind(this);
+
+    this.historyListtenerWasAttached = false;
+
+    UserStore.listen(this._onChange);
+  }
+
+  componentWillUnmount() {
+    UserStore.unlisten(this._onChange);
+  }
+
+  componentWillReceiveProps() {
+    if(!this.historyListtenerWasAttached && this.props.history) {
+      this.props.history.listen(this._onChange);
+      this.historyListtenerWasAttached = true;
+    }
+  }
+
+  _onChange() {
+    this.setState({ currentUser: UserStore.state.currentUser });
+    var self = this;
+    setTimeout(function() {
```

Comment:
timeout is need here becouse `props.location.pathname` dosn't update on time and we end up with redirections loop.

I will try to use arrow function here.